internal/backends/testing/search_api.rs

```diff
@@ -494,6 +494,150 @@ impl ElementHandle {
+            return;
+        };
+
+        let button = i_slint_core::platform::PointerEventButton::Left;
+
+        let Some(item) = self.item.upgrade() else { return };
+        let Some(window_adapter) = item.window_adapter() else { return };
+        let window = window_adapter.window();
+
+        let item_pos = self.absolute_position();
+        let item_size = self.size();
+        let position = LogicalPosition::new(
+            item_pos.x + item_size.width / 2.,
+            item_pos.y + item_size.height / 2.,
+        );
+
+        window.dispatch_event(i_slint_core::platform::WindowEvent::PointerMoved { position });
+        window.dispatch_event(i_slint_core::platform::WindowEvent::PointerPressed {
+            position,
+            button,
+        });
+
+        wait_for(single_click_duration).await;
+
+        window.dispatch_event(i_slint_core::platform::WindowEvent::PointerReleased {
+            position,
+            button,
+        });
+        window.dispatch_event(i_slint_core::platform::WindowEvent::PointerPressed {
+            position,
+            button,
+        });
+
+        wait_for(single_click_duration).await;
+
+        window_adapter.window().dispatch_event(
+            i_slint_core::platform::WindowEvent::PointerReleased { position, button },
+        );
+    }
+
+    /// Simulates a mouse right click on the element at its center point.
+    pub async fn right_click(&self) {
+        self.click(
+            i_slint_core::platform::PointerEventButton::Right,
+            std::time::Duration::from_millis(50),
+        )
+        .await;
+    }
+
+    async fn click(
+        &self,
+        button: i_slint_core::platform::PointerEventButton,
+        duration: std::time::Duration,
+    ) {
+        let Some(item) = self.item.upgrade() else { return };
+        let Some(window_adapter) = item.window_adapter() else { return };
+        let window = window_adapter.window();
+
+        let item_pos = self.absolute_position();
+        let item_size = self.size();
+        let position = LogicalPosition::new(
+            item_pos.x + item_size.width / 2.,
+            item_pos.y + item_size.height / 2.,
+        );
+
+        window.dispatch_event(i_slint_core::platform::WindowEvent::PointerMoved { position });
+        window.dispatch_event(i_slint_core::platform::WindowEvent::PointerPressed {
+            position,
+            button,
+        });
+
+        wait_for(duration).await;
+
+        window_adapter.window().dispatch_event(
+            i_slint_core::platform::WindowEvent::PointerReleased { position, button },
+        );
+    }
+}
+
+async fn wait_for(duration: std::time::Duration) {
+    enum AsyncTimerState {
+        Starting,
+        Waiting(std::task::Waker),
+        Done,
+    }
+
+    let state = std::rc::Rc::new(std::cell::RefCell::new(AsyncTimerState::Starting));
+
+    std::future::poll_fn(move |context| {
+        let mut current_state = state.borrow_mut();
```

Comment:
```suggestion
    let mut state = AsyncTimerState::Starting;
    std::future::poll_fn(|context| {
        let mut current_state = &mut state; 
```

(The last line is only there to make it work with the rest of your code, but is actually not needed)


But this whole function is quite elegant with poll_fn

Reply:
Hm, that doesn't compile for me, because the state is also needed within the Timer's single_shot closure, hence the Rc and RefCell :-/